api/pkg/utils/cryptography/aes.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 The SODA Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cryptography
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	"io"
+)
+
+type AES struct{}
+
+func NewAES() *AES {
+	return &AES{}
+}
+
+func (*AES) Encrypter(password string, key []byte) (string, error) {
```

Comment:
There are already code for AES Encryption and Decryption at few places mostly in Object storage. Is that not re-usable?

Reply:
Yes, I have referenced the code already exists. But could able to use that because of the hard-coded interfaces. May be We need to refactor the framework to make it available as Utility for all SODA projects.

Reply:
Ok. This similar code is at multiple places and this is not good for maintainability

Reply:
Yes, There are several codes in SODA similar to this which are redundant and is not good for maintainability.
I think this code is much better than the existing code in terms of flexibility of interface and readability. You can consider this point.

Reply:
Ok